src/main/java/org/broadinstitute/hellbender/engine/ProgressMeter.java

```diff
@@ -210,15 +230,17 @@ public void stop() {
         Utils.validate( !stopped, "the progress meter has been stopped already");
         this.stopped = true;
         currentTimeMs = timeFunction.getAsLong();
-        logger.info(String.format("Traversal complete. Processed %d total records in %.1f minutes.", numRecordsProcessed, elapsedTimeInMinutes()));
+        logger.info(String.format("Traversal complete. Processed %d total %s in %.1f minutes.", numRecordsProcessed, recordLabel, elapsedTimeInMinutes()));
```

Comment:
I've added a final call to `printProgress()` in `stop()`, as requested.

Since the existing progress meter class has a notion of "current locus", I think in its current state it only makes sense to use it for traversals over the genome. We could consider generalizing it for other purposes in a future PR, however.